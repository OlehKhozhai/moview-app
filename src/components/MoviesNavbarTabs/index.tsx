import React from 'react';
import cn from 'classnames';
import { useDispatch, useSelector } from 'react-redux';

import MoviesNavbarTabItem from 'components/MoviesNavbarTabItem';
import { TABS } from 'config';
import { setActiveTabAction } from 'redux/actions';
import { activeTabSelector } from 'redux/selectors';
import styles from './styles.module.scss';

type MoviesNavbarTabsProps = {
  className?: string;
};

const MoviesNavbarTabs: React.FC<MoviesNavbarTabsProps> = ({ className }) => {
  const activeTab = useSelector(activeTabSelector);
  const dispatch = useDispatch();

  const handleTabClick = React.useCallback(
    (tab) => {
      dispatch(setActiveTabAction(tab));
    },
    [dispatch]
  );

  return (
    <ul className={cn(styles.root, className)}>
      {TABS.map((tab) => {
        return (
          <MoviesNavbarTabItem
            key={tab}
            title={tab}
            isActive={activeTab === tab}
            onTabClick={handleTabClick}
          />
        );
      })}
    </ul>
  );
};

export default React.memo(MoviesNavbarTabs);
