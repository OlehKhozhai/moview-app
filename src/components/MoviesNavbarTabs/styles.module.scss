@import 'assets/styles/mixins/colors';

.root {
  display: flex;
  align-items: center;
}
