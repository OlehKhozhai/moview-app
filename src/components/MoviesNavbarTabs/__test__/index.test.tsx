import React from 'react';
import { create } from 'react-test-renderer';

import MoviesNavbarTabs from '../index';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn().mockReturnValue({ replace: jest.fn() }),
}));

describe('MoviesNavbarTabs component', () => {
  test('should render MoviesNavbarTabItem component by default', () => {
    const component = create(<MoviesNavbarTabs searchParams="" />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
