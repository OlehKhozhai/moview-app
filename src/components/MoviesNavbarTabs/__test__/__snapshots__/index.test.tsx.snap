// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesNavbarTabs component should render MoviesNavbarTabItem component by default 1`] = `
<ul
  className="root"
>
  <li
    className="root active"
    onClick={[Function]}
  >
    all
  </li>
  <li
    className="root"
    onClick={[Function]}
  >
    documentary
  </li>
  <li
    className="root"
    onClick={[Function]}
  >
    comedy
  </li>
  <li
    className="root"
    onClick={[Function]}
  >
    horror
  </li>
  <li
    className="root"
    onClick={[Function]}
  >
    crime
  </li>
</ul>
`;
